aea/runtime.py

```diff
@@ -258,7 +258,11 @@ def _start(self) -> None:
 
         self._agent.multiplexer.set_loop(asyncio.new_event_loop())
 
-        self._agent.multiplexer.connect()
+        # it is important that 'connect' is before 'start setup'.
+        # some skills might start using the multiplexer queue (e.g. gym skill)
+        # and the interleaving with the multiplexer connection set up
+        # might cause concurrency issues.
+        self._agent._connect()
```

Comment:
So both connect and start setup should not be private;...

Reply:
I think yes, instead. It is not supposed to be a public API.

Reply:
I think yes, instead. It is not supposed to be a public API.

Reply:
I suggest leaving as it is, so we avoid to add too much stuff in one PR, and given that we were already using that approach, although I agree that it is a design smell (as many other things... one step at a time :-) )